robotiq_hande_driver/hardware/include/robotiq_hande_driver/application.hpp

```diff
@@ -0,0 +1,315 @@
+#ifndef ROBOTIQ_HANDE_DRIVER__APPLICATION_HPP_
+#define ROBOTIQ_HANDE_DRIVER__APPLICATION_HPP_
+
+#include <stdint.h>
+#include <unistd.h>
+
+#include "protocol_logic.hpp"
+
+
+namespace robotiq_hande_driver
+{
+
+constexpr auto GRIPPER_CURRENT_SCALE = 0.01;
+constexpr auto MAX_SPEED = 255;
+constexpr auto MAX_FORCE = 255;
+
+/**
+ * @brief This class contains high-level gripper commands and status.
+ */
+class GripperApplication {
+public:
+
+    struct Status {
+        bool is_reset;
+        bool is_ready;
+        bool is_moving;
+        bool is_stopped;
+        bool is_opened;
+        bool is_closed;
+        bool object_detected;
+    };
+
+    struct FaultStatus {
+        bool is_error;
+    };
+
+    GripperApplication();
+
+    ~GripperApplication() {};
+
+    /**
+     * @brief Initializes driver parameters.
+     *
+     * @param gripper_position_min Minimal gripper position in meters.
+     * @param gripper_position_max Maximal gripper position in meters.
+     * @param tty_port Modbus virtual port.
+     * @param baudrate Modbus serial baudrate.
+     * @param parity Modbus serial parity.
+     * @param data_bits Modbus serial data bits.
+     * @param stop_bit Modbus serial stopbit.
+     * @param slave_id Modbus slave id.
+     * @return None.
+     * @note The status should be checked to verify successful execution. An exception is thrown if communication issues occur.
+     */
+    void initialize(double gripper_position_min, double gripper_position_max, std::string tty_port, int baudrate, char parity, int data_bits, int stop_bit, int slave_id) {
```

Comment:
Let's reduce double copying of the string argument

```suggestion
    void initialize(double gripper_position_min, double gripper_position_max, const std::string& tty_port, int baudrate, char parity, int data_bits, int stop_bit, int slave_id) {
```

Reply:
Done.